decidim-proposals/app/views/decidim/proposals/proposals/_vote_button.html.erb

```diff
@@ -1,23 +1,25 @@
-<div id="proposal-<%= proposal.id %>-vote-button">
-  <% if !current_user %>
-    <% if current_settings.votes_blocked? %>
-      <%= action_authorized_button_to :vote, t('.votes_blocked'), proposal_proposal_vote_path(proposal_id: proposal, from_proposals_list: from_proposals_list), class: "card__button button #{vote_button_classes(from_proposals_list)} disabled", disabled: true %>
-    <% else %>
-      <%= action_authorized_button_to :vote, t('.vote'), proposal_proposal_vote_path(proposal_id: proposal, from_proposals_list: from_proposals_list), class: "card__button button #{vote_button_classes(from_proposals_list)}" %>
-    <% end %>
-  <% else %>
-    <% if @voted_proposals ? @voted_proposals.include?(proposal.id) : proposal.voted_by?(current_user) %>
-      <%= action_authorized_button_to :vote, t('.already_voted'), proposal_proposal_vote_path(proposal_id: proposal, from_proposals_list: from_proposals_list), method: :delete, remote: true, data: { disable: true }, class: "card__button button #{vote_button_classes(from_proposals_list)} success" %>
+<% unless proposal.answered? && proposal.rejected? %>
+  <div id="proposal-<%= proposal.id %>-vote-button">
+    <% if !current_user %>
+      <% if current_settings.votes_blocked? %>
```

Comment:
Maybe we should move all this logic into a helper method?

Reply:
hmm.. what about using the `rejected?` method and just adding the `answered?` condition?

Reply:
Like this:

```ruby
      # Public: Checks if the organization has rejected a proposal.
      #
      # Returns Boolean.
      def rejected?
        answered? && state == "rejected"
      end
```